tests/integration/devfile/cmd_devfile_push_test.go

```diff
@@ -70,6 +70,8 @@ var _ = Describe("odo devfile push command tests", func() {
 			helper.CmdShouldPass("odo", "preference", "set", "Experimental", "true")
 
 			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs"), context)
+			cmpName := helper.RandString(6)
+			helper.CmdShouldPass("odo", "create", "nodejs", "--context", context, "--namespace", namespace, cmpName)
```

Comment:
This command shouldn't pass based on the design https://github.com/openshift/odo/issues/2789 we discussed before.

```
If devfile.yaml is present
Case 1: odo create <component type> -> Throw errors and tell user to delete the existing devfile.yaml then run odo create again
```
If you want to specify the component type, you have to delete the existing devfile.yaml as component type is used for downloading the devfile.yaml. Or you can run odo create without component type if you want to leave the devfile.yaml

Reply:
This PR is not going to handle the changes to the create command. The create validation should be handled by your PR.

Reply:
I'm not sure you get my point, It doesn't matter whose PR handles this issue, it matters your test case is WRONG here based on the design (I mean your test case should detect the errors instead of letting it pass), if you decides to remove the devfile.yaml existence check here https://github.com/openshift/odo/pull/2762/files#diff-58fe314b4e3f9120402717ab8c2b3049R588, you have to handle the corresponding behaviors correctly. Or you shouldn't remove that and wait for my PR to handle that, it totally depends on you.

The key point is with you changes, it can causes the regression of this use case below:
```
If user has nodejs devfile.yaml in the workspace directory but user runs odo create maven, it can still pass but that's not expected
```

Reply:
Also keep in mind by design currently there are no required/mandatory arguments, we have interactive mode handles the use case.

Reply:
As we discussed in the meeting, the validation should be handled in a separate PR. 
The `componentType` is a required argument in master branch. Since the validation is missing, you cannot expect the test case to meet the behavior like that. 
But I don't mind take that issue and doing the whole validation in this PR.  You can assign the new issue to me, I will do the entire validation in the PR.

Reply:
We do have interactive mode, which means even for today's master branch component type is also not required (same design as odo1), you can give a try. 

But anyway I already opened a PR https://github.com/openshift/odo/pull/2798 to cover all cases of https://github.com/openshift/odo/issues/2789, and already asked @kadel review that, you can talk with him if you want to take over that

Reply:
Just tried, if would fail without specifying any componentType:
```
 ✗  Checking if the specified component type is supported devfile component type [99789ns]

Please run 'odo catalog list components' for a list of supported devfile component types
panic: runtime error: invalid memory address or nil pointer dereference
```